packages/frontend/src/notebook/notebook_cell.tsx

```diff
@@ -147,13 +157,27 @@ export function NotebookCell(props: {
                     return isCellDragData(source.data);
                 },
                 getData({ input }) {
-                    const data = createCellDragData(props.cellId);
+                    const data = createCellDragData(props.cellId, props.index);
                     return attachClosestEdge(data, {
                         element: rootRef,
                         input,
                         allowedEdges: ["top", "bottom"],
                     });
                 },
+                onDragEnter: (args) => {
```

Comment:
Not actually important, but there's syntactic sugar which would let you write this first line as `onDragEnter(args) {`